source/Metadata/Javadoc.cpp

```diff
@@ -17,19 +18,22 @@
 namespace clang {
 namespace mrdox {
 
-template<class T>
-concept a_Node =
-    std::is_copy_constructible_v<T> &&
-    std::three_way_comparable<T>;
+//------------------------------------------------
 
-static_assert(a_Node<doc::Node>);
-static_assert(a_Node<doc::Text>);
-static_assert(a_Node<doc::StyledText>);
-static_assert(a_Node<doc::Block>);
-static_assert(a_Node<doc::Paragraph>);
-static_assert(a_Node<doc::Param>);
-static_assert(a_Node<doc::TParam>);
-static_assert(a_Node<doc::Code>);
+template<typename T, typename U>
+static
+auto
+move_to(
+    doc::List<T>& dst,
+    doc::List<U>& src,
+    typename doc::List<U>::iterator it)
+{
+    auto elem = std::move(*it);
+    it = src.erase(it);
+    dst.emplace(dst.end(),
+        static_cast<T*>(elem.release()));
```

Comment:
why not `std::move(*it)`?

Reply:
`dst` may be a `std::vector<std::unique_ptr<U>>` where `U` is derived from `T`